app/src/main/java/mozilla/lockbox/presenter/RoutePresenter.kt

```diff
@@ -91,4 +93,14 @@ class RoutePresenter(
 
         return null
     }
+
+    override fun onPause() {
+        super.onPause()
+        Dispatcher.shared.dispatch(LifecycleAction.Background)
+    }
+
+    override fun onResume() {
+        super.onResume()
+        Dispatcher.shared.dispatch(LifecycleAction.Foreground)
+    }
```

Comment:
This might work for now, however, it's extremely extremely brittle:

 * in the event of a configuration change (for example but not limited to: orientation change, car mode, night mode etc), `onPause` will be fired immediately followed by `onResume`.
 * when we add more activities (we will do shortly), the same will happen.

Suggest: 

 * we have an `ApplicationPresenter`, which fires off these events.
 * we use a [`ProcessLifecycleOwner`][1] and `LifecycleObserver` which does the OS specific detection, and hands off to the `ApplicationPresenter`. The [`ProcessLifecycleOwner`][1] is designed for this exact use case of detecting the app going into background and coming back in to foreground.

[1]: https://developer.android.com/reference/android/arch/lifecycle/ProcessLifecycleOwner